### PR TITLE
Update unitig-caller to v1.3.0

### PR DESCRIPTION
**Pull request by johnlees**
Updates unitig-caller to v1.3.0 (and fixes the issues in https://github.com/bacpop/unitig-caller/issues/13 and https://github.com/bioconda/bioconda-recipes/issues/36884)

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by jmarshall** [comment]
I see the unitig-caller source code now bundles a copy of sdsl-lite-3, presumably because that has better FM index functionality than the older sdsl-lite-2 (which is packaged in conda-forge).

Conda packaging of sdsl-lite-3 appears to be an ongoing saga (cf conda-forge/staged-recipes#17965). It's unfortunate for a bioconda package to bundle an external library; however as it is a header-only library there is no difference either way at runtime and there would be no runtime dependency, so it is relatively immaterial as a compatibility matter for the bioconda package. There is a maintenance headache for upstream, but it doesn't really affect the bioconda packaging IMHO.

File: recipes/unitig-caller/meta.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ package:
 
 source:
   url: https://github.com/bacpop/{{ name|lower }}/archive/v{{ version }}.tar.gz
-  sha256: 61ff47df9c5339bd4cd7f6c4c7073ccb82730b645388e07369730c7c39309f33
+  sha256: 9059db7042c9b6ae240974fc474bd51d092d1f8795009a4e9e969abfc293c007
```
Comment (jmarshall) [inline recipes/unitig-caller/meta.yaml:10]: There's a reason Git makes it difficult to change tags that have already been published, you know… :smile:

**Comment by jmarshall** [comment]
Did you follow the link in the macOS error message?

> /usr/local/miniconda/envs/bioconda/conda-bld/unitig-caller_1664884629378/work/src/map_bindings.cpp:56:22: error: 'path' is unavailable: introduced in macOS 10.15 - see https://conda-forge.org/docs/maintainer/knowledge_base.html#newer-c-features-with-old-sdk
>   std::filesystem::path p(file);

It would appear that it is safe to work around this by adding `-D_LIBCPP_DISABLE_AVAILABILITY` to $CXXFLAGS. So it may be worth one more try at the macOS build, adding that. Or were there other errors of a different category too?

Otherwise, this is looking about right for a Linux-only packaging. If you can't bring up the macOS build, you should omit the patches. I'm not very excited about the `UC_INSTALL` fingers reaching into the upstream build scripts, but I see that's already there in the new release and that _CMakeLists.txt_ also looks at $CONDA_PREFIX…

**Comment by johnlees** [comment]
> It would appear that it is safe to work around this by adding `-D_LIBCPP_DISABLE_AVAILABILITY` to $CXXFLAGS. So it may be worth one more try at the macOS build, adding that. Or were there other errors of a different category too?

No I missed this, thanks for pointing it out! A little painful with the [required link time library](https://releases.llvm.org/8.0.0/projects/libcxx/docs/UsingLibcxx.html#using-libc-experimental-and-experimental) too, but finally all passing

> Otherwise, this is looking about right for a Linux-only packaging. If you can't bring up the macOS build, you should omit the patches. I'm not very excited about the `UC_INSTALL` fingers reaching into the upstream build scripts, but I see that's already there in the new release and that *CMakeLists.txt* also looks at $CONDA_PREFIX…

One day I'll work out a CMakeLists that works for linux and OS X inside and outside of a conda environment/install, but I have stuggled without a helper env variable so far...

**Comment by dpryan79** [comment]
@bioconda-bot please fetch artifacts

**Review by dpryan79 — APPROVED** [review]

**Comment by dpryan79** [comment]
@bioconda-bot please merge

**Pull request merged**